pkg/sources/source.go

```diff
@@ -146,14 +149,26 @@ func (sp *SourceProcessor) Start(ctx context.Context) error {
 				if err != nil {
 					return err
 				}
-				writers = append(writers, writer)
+				bufferWriters = append(bufferWriters, writer)
 			}
+			writersMap[e.To] = bufferWriters
 		}
 	default:
 		return fmt.Errorf("unrecognized isb svc type %q", sp.ISBSvcType)
 	}
 	var sourcer Sourcer
 	var readyChecker metrics.HealthChecker
+	// Populate shuffle function map
```

Comment:
> there was a bug in source vertex code, if there was a reduce vertex after the source vertex we were not shuffling the messages.

@yhl25 , I don't think it is a bug. Before this PR, the contract of shuffling at source is that if the source vertex has a UDTransformer, then we shuffle using `getTransformerGoWhereDecider`, this is because no UDTransformer means no map operation, hence we don't expect an immediate reduce as next vertex simply because we haven't assign any keys to source messages. I have two questions:

1. With this change, we indicate that if the source doesn't have a UDTransformer, we can still have a reduce vertex immediately after it. Is it correct?

2. With multi-partitioned edge, we are going to support shuffling when the next vertex is a map. Assuming that's in place, would we be able to generalize `sp.getTransformerGoWhereDecider(shuffleFuncMap)` and `sp.getSourceGoWhereDecider(shuffleFuncMap)` into one?